src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -306,66 +323,127 @@ subroutine init_atm_static(mesh, dims, configs)
+     ! pixel by using a KD search and assign its value to that cell.
+     !
+     do while (.not. mgr % is_stack_empty())
+         tile => mgr % pop_tile()
+
+         if (tile % is_processed) then
+             cycle
+         end if
+
+         call mpas_log_write('Processing tile: '//trim(tile%fname))
+
+         all_pixels_mapped_to_halo_cells = .true.
+
+         do j = 1 + tile_bdr, tile_ny + tile_bdr, 1
+            do k = 1 + tile_bdr, tile_nx + tile_bdr, 1
+
+               tval = tile % tile(k, j, 1)
+
+               call mgr % tile_to_latlon(tile, j, k, lat_pt, lon_pt)
+               call mpas_latlon_to_xyz(xPixel, yPixel, zPixel, sphere_radius, lat_pt, lon_pt)
+               call mpas_kd_search(tree, (/xPixel, yPixel, zPixel/), res)
+
+               !
+               ! For all but the outermost boundary cells, we can safely assume that the nearest
+               ! model grid cell contains the pixel (else, a different cell would be nearest)
+               !
+               if (bdyMaskCell(res % id) < nBdyLayers) then
+                  ter_integer(res % id) = ter_integer(res % id) + int(tval, kind=I8KIND)
+                  nhs(res % id) = nhs(res % id) + 1
+
+                  if (res % id <= nCellsSolve) then
+                      all_pixels_mapped_to_halo_cells = .false.
+                  end if
+
+               ! For outermost boundary cells, additional work is needed to verify that the pixel
+               ! actually lies within the nearest cell
+               else
+                  if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                   nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+
+                     ter_integer(res % id) = ter_integer(res % id) + int(tval, kind=I8KIND)
+                     nhs(res % id) = nhs(res % id) + 1
+
+                     if (res % id <= nCellsSolve) then
+                         all_pixels_mapped_to_halo_cells = .false.
+                     end if
+                  end if
+               end if
+            end do
+        end do
+
+        tile % is_processed = .true.
+
+        !
+        ! If a single pixel value maps to an owned cell (i.e. <= nCellsSolve) then
+        ! it is possible that the neighboring tiles might contain pixels that map
+        ! to this process' compute cells, so add them to the stack.
+        !
+        if (.not. all_pixels_mapped_to_halo_cells) then
+            ierr = mgr % push_neighbors(tile)
+            if (ierr /= 0) then
+                call mpas_log_write("Error pushing the tile neighbors of: "//trim(tile%fname), messageType=MPAS_LOG_CRIT)
+            end if
+        end if
+     end do
  end do
 
- do iCell = 1,nCells
-    ter(iCell) = ter(iCell) / real(nhs(iCell))
+ do i = 1, nCells
+    ter(i) = real(ter_integer(i), kind=RKIND) / real(nhs(i), kind=RKIND)
```

Comment:
For current datasets I don't think it would make much difference, but because its easy enough to do so, we could type-convert the integer values to `R8KIND` rather than `RKIND` before performing the division.